acp/src/commonMain/kotlin/com/agentclientprotocol/client/Client.kt

```diff
@@ -215,7 +225,27 @@ public class Client(
         return completableDeferred.getCompleted()
     }
 
-    private suspend fun getSessionOrThrow(sessionId: SessionId): ClientSessionImpl = (_sessions.value[sessionId] ?: acpFail("Session $sessionId not found")).await()
+    private suspend fun getSessionOrThrow(sessionId: SessionId): ClientSessionImpl {
+        _sessions.value[sessionId]?.let {
```

Comment:
This part is somewhat redundant, i.e. the whole `completableDeferred` could be removed and replaced with just a map to the session. In case of only one session initialization the behavious will be exactly the same, because the _currentlyInitializingSessionsCount will become 0 right after the `sessionDeferred.complete(session)`.

And case of simultanious session initialization seems to be rare (i.e. we shouldn't optimize for it)